decidim-core/lib/decidim/amendable.rb

```diff
@@ -81,46 +81,45 @@ def amendable(fields: nil, form: nil)
       end
     end
 
+    # Returns the fields that can be amended.
     def amendable_fields
       self.class.amendable_options[:fields]
     end
 
+    # Returns the form used for the validation and creation of the emendation.
     def amendable_form
       self.class.amendable_options[:form].constantize
     end
 
+    # Returns the ActiveRecord class name of the resource.
     def amendable_type
       resource_manifest.model_class_name
```

Comment:
I'm thinking, isn't this method kind of dumb? We could do the same by calling just `#class` and we wouldnt need to call `#amendable_type.constantize` every time...

Reply:
Could you elaborate? I don't have all the context so I'm not sure where we're calling `amendable_type.constantize`.

Reply:
I've made a search on the project an found 7 matches (thought there were more). In short, we are using `amendable_type.constantize` as an alternative to `class`...

Some examples are:
```ruby
# Passing the class of the amendable resource as argument.
Decidim.traceability.perform_action!(
  "publish",
  amendable.amendable_type.constantize,
  amender,
  visibility: "public-only"
)

# Instantiating an object of the amendable resource class.
emendation = amendable.amendable_type.constantize.new(form.emendation_params)
```
I could make a different PR to refactor it if deemed appropriate.

Reply:
I don't quite follow how we could use `class` 😅 (`model_class_name` returns a `String`)

Reply:
You're right, and sometimes we want just that, but then again, wouldn't it be easier to call `class.to_s` if we want the name of the class as `String`?

Reply:
At the resource manifest we expect to have it as a String, not a Class. This is because if we wrote classnames instead of strings, then we have problems with autoloading and reloading during development, so anywhere where we need to refer to a class we always do it as a String.

Reply:
Of course!

I meant that the method `Decidim::Amendable#amendable_type` is in practice a convoluted way of invoking `class` on the instance. I was not questioning at all the `Decidim::ResourceManifest`' `model_class_name` attribute.

Sorry for the misunderstanding :see_no_evil:

Reply:
Ohh, you mean adding `constantize` to `amendable_type`? That's OK!

Reply:
Well, I was thinking of getting rid of that instance method altogether, and just call `class` or `class.to_s` on the amendable resource when needed.